Dockerfile

```diff
@@ -0,0 +1,19 @@
+# For local development purposes only
+
+FROM httpd:alpine
+
+RUN rm -r /usr/local/apache2/htdocs/*
+COPY /build /usr/local/apache2/htdocs/
```

Comment:
I made a decision here that I'm not 100% confident on. 

This is copying what's in your build folder and running the server from that copy. 

I could have instead made it run with a mapped volume, so that it is actually running off of your build folder directly. In which case you could change things in there, and you'd see them in the running instance. 

I chose to do the copy instead of the mapped volume for two reasons: 

1. Part of the configuration is removing the SSL requirement, which means editing the .htaccess file, and I don't like the idea of editing your build folder directly. 
2. I _think_ I'd need to restart Apache on the running instance after making changes to .htaccess. That isn't itself a reason not to choose this method, but it makes this method less appealing.

Reply:
Oh, and I should add that the down-side to running from a copy instead of the build folder directly is that if I want to make changes to my htaccess and see it reflected in the running instance, I need to rebuild with Docker, which makes it a little tedious. 

But with having to restart Apache after making changes when running directly from build, I think it's a bit tedious either way.

Reply:
I don't think you can get out of rebuilding.  I'm ultimately unsure about restarting Apache. What is your goal here? Ideally, avoid rebuilding, restarting, and modifying the build folder? Or maybe minimize tediousness? 

In our build workflow, we do the following steps (which you may also not like, but have tolerated, knowingly or unknowingly):

```
- name: Build
   run: npm run build
   env:
        NODE_OPTIONS: --max_old_space_size=4096
- name: Remove https redirect
   run: |
        sed -i '/SERVER_PORT/d;/SERVER_NAME/d' build/.htaccess
- name: Run web server
    run: |
       docker run -d --name webserver -v "$PWD/build":/app -p 8888:8080 bitnami/apache:latest
       echo "container_ip=$(docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' webserver)" >> $GITHUB_ENV
```

So that's build > _modify_ build file > run web server. 

We are running into a situation where compromising (and documenting said compromise) is the best thing to do.

@menski thoughts here?

Reply:
I may also not fully understand this because of my mediocre understanding of Apache.

Reply:
I'm thinking specifically of this feedback loop, which is the reason one would want to run this server locally:

* .htaccess redirects are failing in CI but we can't figure out why
* I start up the server locally
* I enter a URL in the address bar that I think should redirect
* It goes to the wrong place
* I want to edit my .htaccess to make an adjustment
* I enter the URL in the address bar 
* ➿ 

It'd be great if there was nothing between editing the .htaccess and entering something in the address bar...but I think regardless of which of the methods I described above, there's a step where I have to bounce something before I can see the changes.

Reply:
> In our build workflow, we do the following steps (which you may also not like, but have tolerated, knowingly or unknowingly):

wowww I feel like I must have known that, but somehow have not noticed or remembered it.

Reply:
Do you have an issue in the backlog for the follow-up work?